### DISCUSSION TITLE
Virtual IP assigned has different mask from what is specified in swanctl.conf pool section

### DISCUSSION THREAD

**Discussion by Dyadya-Sasha** [strongswan/strongswan#783] [Q&A]
Hello everybody!
I am new in VPN networks but try to set it up.
I am stuck a bit by assigning virtual IP with wrong mask.
```
connections {
 
	ios {
		version = 2
		local_addrs = 192.168.0.111
        proposals = aes192gcm16-aes128gcm16-prfsha256-ecp256-ecp521,aes192-sha256-modp3072,aes256gcm16-sha384-prfsha384-ecp384,aes256-aes128-sha256-sha1-modp3072-modp2048-modp1024,default
		pools = xvpn
		local {
			auth = psk
			id = STRONGSWAN_SAN
			}
		remote {
			auth = psk
			}
		children {
		ios {
            esp_proposals = aes192gcm16-aes128gcm16-prfsha256-ecp256-modp3072,aes192-sha256-ecp256-modp3072,aes256-aes128-sha256-sha1-modp3072-modp2048-modp1024,default
			}
	}
	}
}


pools {
	xvpn {
		addrs = 10.0.3.0/24
	}

}

secrets {
    eap-xakup {
        secret = qweqwe
		id = xakup 
   		}
	ike-qwe {
		id = carol@yandex.ru	
		secret = qweqwe
		}
} 
```

Initiator connects to responder and gets virtual IP 10.0.3.1 BUT with mask 32, inspite of 24 in config file

```
ios: #2, ESTABLISHED, IKEv2, aeb322e989d06ef2_i db5a324596679a28_r*
  local  'STRONGSWAN_SAN' @ 192.168.0.111[4500]
  remote 'carol@yandex.ru' @ 192.168.0.169[4500] [10.0.3.1]
  AES_CBC-256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
  established 11s ago, rekeying in 14351s
  ios: #2, reqid 1, INSTALLED, TUNNEL, ESP:AES_CBC-256/HMAC_SHA2_256_128
    installed 11s ago, rekeying in 3319s, expires in 3949s
    in  c2e155fb,      0 bytes,     0 packets
    out 05fe28b9,      0 bytes,     0 packets
    local  192.168.0.111/32
    remote 10.0.3.1/32
```

I googled almost all around and didnt find an answer. 
Could you please explain what and why it happens?

**Comment by tobiasbrunner** [comment]
> Initiator connects to responder and gets virtual IP 10.0.3.1 BUT with mask 32, inspite of 24 in config file

What you define in the `pools` section defines the subnet or range from which virtual IPs are assigned to clients, each of which gets a single address from that address pool (i.e. `/32` or `/128`). See [VirtualIP](https://wiki.strongswan.org/projects/strongswan/wiki/VirtualIp).

**Reply by Dyadya-Sasha** [reply]
> 

But it must be assigned from /24, not from /32. Why 32?

**Reply by tobiasbrunner** [reply]
Not sure what you mean, but that address _is_ assigned from the `/24` subnet. If a second client connects, it will receive `10.0.3.2`, the next one `10.0.3.3` and so on until the last new client will receive `10.0.3.254` (if all clients are concurrently active, no addresses will be available, so the next new client won't be able to connect, otherwise, offline leases will get reassigned to new clients).

**Reply by Dyadya-Sasha** [reply]
Yes, but I would think that /24 is netmask. However, when I connect from windows, I see new interface with netmask 255.255.255.255. Why not 255.255.255.0?

**Reply by tobiasbrunner** [reply]
Why would you think that? And what effect would you expect that to have on the Windows client exactly?  By the way, you could send an `INTERNAL_IP4_NETMASK` attribute to clients if you added `netmask = 255.255.255.0` to the pool you configured, but I've no idea if Microsoft clients (or any other) actually use that.

**Reply by Dyadya-Sasha** [reply]
Understood! Thanks for your help, Tobias!

**Reply by storm37000** [reply]
What if i want 2 clients connected to the strongswan server to be able to act like they are on a LAN together but with no internet routing, nat, or firewalls needed (split tunnel)?
